### DISCUSSION TITLE
Input/Output linting prevents usage of Environment Variables

### DISCUSSION THREAD

**Discussion by AndreasBergmeier6176** [redpanda-data/connect#3048] [Q&A]
When having Linting like in: https://github.com/redpanda-data/connect/blob/5a64fed7c09746f1591d26ce350f2787ca6cae62/internal/impl/kafka/input_kafka_franz.go#L61

it forces Unit-Testing to fail since at lint time, the Environment Variables are not yet set and then in turn the lint check fails.
This change broke some of our pipelines recently.

**Comment by mihaitodor** [comment]
Hey @AndreasBergmeier6176, thanks for raising that! Would it be possible for you to set a default value for your environment variables? For example `${CONSUMER_GROUP:whatever}` (you can also do `${FOOBAR:}` if an empty string is a valid default value)? Otherwise, it would be really difficult to come up with a selective linting scheme for tests which is somehow different than the linting scheme used outside of tests.

LE: Moving to a discussion as per #2026.

**Comment by AndreasBergmeier6176** [comment]
You are correct that technically introducing a default value for an environment variable could fix the linting.
That being said, I think it semantically in the scenario of Kafka Consumer Groups is a BAD IDEA™.
From experience we often have to deal with Kafka Servers where consumer group names are not tested and thus once you screw up the naming (by falling back to the default) it might be a long time until someone notices that you have been using `whatever` for half a year.
Thus, I will try now to first introduce the Environment Variables when calling linting...

**Reply by mihaitodor** [reply]
Indeed, I don't have a great suggestion there. Happy to revise the linting rule if need be. Here's where that change originated from: https://github.com/redpanda-data/connect/pull/2972

**Comment by AndreasBergmeier6176** [comment]
Just defining the Environment Variable when running lint is sufficient though - so fine for me.
